pkg/cloud/aws/assets/validating-admission-policy.yaml

```diff
@@ -0,0 +1,19 @@
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: "system:openshift:cloud-controller-manager:cloud-provider-aws"
+spec:
+  failurePolicy: Fail
+  matchConstraints:
+    resourceRules:
+    - apiGroups:   [""]
+      apiVersions: ["v1"]
+      operations:  ["UPDATE"]
+      resources:   ["services"]
+  validations:
+    - expression: |
+        (has(object.objectMeta.annotations) && 'service.beta.kubernetes.io/aws-load-balancer-type' in object.objectMeta.annotations) ==
+          (has(oldObject.objectMeta.annotations) && 'service.beta.kubernetes.io/aws-load-balancer-type' in oldObject.objectMeta.annotations) &&
+          object.objectMeta.annotations['service.beta.kubernetes.io/aws-load-balancer-type'] == oldObject.objectMeta.annotations['service.beta.kubernetes.io/aws-load-balancer-type']
```

Comment:
I'm not 100% following how this works, so this is more for my understanding - shout if I'm wrong: 

We assert that our old object and our new object have both got the annotation matching `service.beta.kubernetes.io/aws-load-balancer-type`

then we check the value of the annotation remains equal? i.e does not change

```
object.objectMeta.annotations['service.beta.kubernetes.io/aws-load-balancer-type'] == oldObject.objectMeta.annotations['service.beta.kubernetes.io/aws-load-balancer-type']
```
If this annotation were being changed, we would fail the validation - so we  would fail that update, disasllowing changing it's value.

Or am I missing something?

Reply:
> We assert that our old object and our new object have both got the annotation matching service.beta.kubernetes.io/aws-load-balancer-type. We need this because otherwise the controller would not be able to set the value initially?

You're close but missing one nuance, we check that whether "they have the annotation or not" is equal. If one has the annotation and the other does not, then the user changed something, and we aren't allowing that here.

Reply:
Ahh gotcha! Makes sense!